openmdao/docs/openmdao_book/features/building_blocks/drivers/analysis_driver.ipynb

```diff
@@ -0,0 +1,580 @@
+    "\n",
+    "AnalysisDriver takes sample values for any input in the model, as well as implicit outputs.\n",
+    "Setting the value of an explicit output in the model would have no effect, as the value would be overwritten\n",
+    "upon the execution of the model.\n",
+    "\n",
+    "Unlike DOEDriver, it also allows units and indices to be specified for the variables in each sample.\n",
+    "Essentially, any argument to `set_val` can be provided as part of the sample data.\n",
+    "\n",
+    "DOEDriver also realizes all of the design points to be analyized, while AnalysisDriver's generators do this in a lazy way, which could potentially save some memory when running large data sets.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generating Cases\n",
+    "\n",
+    "Creating a sequence of more than a handful of sample cases can be daunting as the number of model inputs increases.\n",
+    "In addition, exceedingly large numbers of cases for large models may begin to tax memory resources if they are all kept in memory.\n",
+    "To alleviate these issues, samples for AnalysisDriver can be provided by a lazy pythonic generator descended from an AnalysisGenerator.\n",
+    "\n",
+    "There are currently three types of AnalysisGenerator included with OpenMDAO\n",
+    "\n",
+    "- ZipGenerator\n",
+    "- ProductGenerator\n",
+    "- CSVGenerator\n",
+    "\n",
+    "## ZipGenerator \n",
+    "\n",
+    "ZipGenerator takes a dictionary that is similar to those given as a sample, except it provids all values to be assumed for each variable.\n",
+    "It then uses python's `zip` function to create a sample for each value specified.\n",
+    "\n",
+    "For instance\n",
+    "\n",
+    "```\n",
+    "ZipGenerator({'x': {'vals': [0.0, 1.0, 2.0], units=None},\n",
+    "              'y': {'vals': [4.0, 5.0, 6.0], units=None})\n",
+    "```\n",
+    "\n",
+    "would generate three sample cases:\n",
+    "\n",
+    "```\n",
+    "[{'x': {'val': 0.0, 'units': None}, 'y': {'val': 4.0, 'units': None}},\n",
+    " {'x': {'val': 1.0, 'units': None}, 'y': {'val': 5.0, 'units': None}},\n",
+    " {'x': {'val': 2.0, 'units': None}, 'y': {'val': 6.0, 'units': None}}]\n",
+    "```\n",
+    "\n",
+    "Note that the number of values given for each variable to be sampled must be the same.\n",
+    "\n",
+    "## ProductGenerator \n",
+    "\n",
+    "ProductGenerator, like ZipGenerator, takes a dictionary that provides values and optionally units and indices for\n",
+    "each variable to be sampled.\n",
+    "It uses Python's `itertools.product` to produce every possible combination of the values of sampled variables.\n",
+    "\n",
+    "For instance\n",
+    "\n",
+    "```\n",
+    "ProductGenerator({'x': {'vals': [0.0, 1.0, 2.0], units=None},\n",
+    "                  'y': {'vals': [4.0, 5.0, 6.0], units=None})\n",
```

Comment:
I know you mention it later that variables can have different numbers of values, but might be even better to actually have x and y have different numbers of vals in this example.